### PR TITLE
get hosted-zone name and id where hosted-zone-name not equal to master-zone name

### PR DESCRIPTION
**Pull request by pwyborn**
Hi @here could someone please review these changes re set_active_stack (to cater for where the hosted_zone is not equal to master_zone names)

### CODE REVIEW

File: bootstrap_cfn/fab_tasks.py [hunk 1]
```diff
@@ -496,6 +495,30 @@ def get_zone_id():
...
+        record = "{name}.{hosted_zone}".format(**elb)
+        zone_id = r53_conn.get_hosted_zone_id(elb['hosted_zone'])
+        return elb['hosted_zone']
+
+@task
+def get_hosted_id():
+    cfn_config = get_config()
+
+    r53_conn = get_connection(R53)
+    import pdb;
```
Comment (yufangzhang) [inline bootstrap_cfn/fab_tasks.py]: import pdb; XD could we remove these?

File: bootstrap_cfn/fab_tasks.py [hunk 2]
```diff
@@ -496,6 +495,30 @@ def get_zone_id():
                 "for zone name '%s'...", zone_id, zone_name)
     return zone_id
 
+@task
+def get_hosted_name():
+    cfn_config = get_config()
+
+    r53_conn = get_connection(R53)
+    import pdb;
+    for elb in cfn_config.data['elb']:
```
Comment (yufangzhang) [inline bootstrap_cfn/fab_tasks.py]: you may want to check if cfn_config.data has KeyError when 'elb' is not in yaml by adding try and except, or cfn_config.data.get('elb') returns none if the key doesn't exist.

**Comment by pwyborn** [comment]
Hi @yufangzhang - can you please re-check (changes made as per your suggestions - thanks)

File: bootstrap_cfn/fab_tasks.py [hunk 3]
```diff
@@ -496,6 +494,38 @@ def get_zone_id():
...
+                raise CfnConfigError("No hosted_zone in yaml, unable to find hosted zone")
+            logger.info("fab_tasks::get_hosted_name: Found hosted zone name '%s' in config...", elb['hosted_zone'])
+            if elb['hosted_zone'].endswith('.'): 
+                return elb['hosted_zone'][:-1]
+            else:
+                return elb['hosted_zone']
+    except KeyError:
+        raise CfnConfigError("No elb in yaml, unable to find elb")
+
+def get_hosted_id():
```
Comment (yufangzhang) [inline bootstrap_cfn/fab_tasks.py]: @pwyborn get_hosted_id() and get_hosted_name look exactly the same to me?

File: bootstrap_cfn/fab_tasks.py [hunk 4]
```diff
@@ -496,6 +494,38 @@ def get_zone_id():
...
+    cfn_config = get_config()
+    r53_conn = get_connection(R53)
+    try:
+        for elb in cfn_config.data['elb']:
+            try:
+                record = "{name}.{hosted_zone}".format(**elb)
```
Comment (yufangzhang) [inline bootstrap_cfn/fab_tasks.py:521]: don't think you need 'record'? is it used anywhere?

File: bootstrap_cfn/fab_tasks.py [hunk 5]
```diff
@@ -496,6 +494,38 @@ def get_zone_id():
                 "for zone name '%s'...", zone_id, zone_name)
     return zone_id
 
+def get_hosted_name():
+    cfn_config = get_config()
+    r53_conn = get_connection(R53)
+    try:
+        for elb in cfn_config.data['elb']:
+            try:
+                record = "{name}.{hosted_zone}".format(**elb)
```
Comment (yufangzhang) [inline bootstrap_cfn/fab_tasks.py:503]: same here

File: bootstrap_cfn/fab_tasks.py [hunk 6]
```diff
@@ -496,6 +494,38 @@ def get_zone_id():
...
+                zone_name = elb['hosted_zone']
+            except KeyError:
+                raise CfnConfigError("No hosted_zone in yaml, unable to find hosted zone")
+            logger.info("fab_tasks::get_hosted_name: Found hosted zone name '%s' in config...", elb['hosted_zone'])
+            if zone_name.endswith('.'): 
+                return zone_name[:-1]
+            else:
+                return zone_name
```
Comment (yufangzhang) [inline bootstrap_cfn/fab_tasks.py:511]: for multiple ELBs, this returns the first elb['hosted_zone']. I think we may want to return specific one that matches elb['name']

**Comment by yufangzhang** [comment]
similar PR 268 has been reviewed and merged. this is not needed any more...

**Pull request closed without merging**